ocean-client/src/programs/vault-maxi-program.ts

```diff
@@ -333,102 +508,201 @@ export class VaultMaxiProgram extends CommonProgram {
+        if (token) {//removing exposure: keep wallet clean
+            if (this.isSingleMint) {
+                collateralTokens.push(token)
+            } else {
+                paybackTokens.push(token)
+            }
         }
-        token = tokens.get(this.settings.LMToken)
-        if (token) { //reducing exposure: keep wallet clean
+        token = tokens.get(this.assetA)
+        if (token) { //removing exposure: keep wallet clean
             paybackTokens.push(token)
         }
 
-        if (await this.paybackTokenBalances(paybackTokens, telegram)) {
+        if (await this.paybackTokenBalances(paybackTokens, collateralTokens, telegram)) {
             await telegram.send("done removing exposure")
             return true
         }
         return false
     }
 
-    private async paybackTokenBalances(addressTokens: AddressToken[], telegram: Telegram, prevout: Prevout | undefined = undefined): Promise<boolean> {
-        let paybackTokens: TokenBalance[] = []
-        addressTokens.forEach(addressToken => {
-            paybackTokens.push({ token: +addressToken.id, amount: new BigNumber(addressToken.amount) })
-        })
-        console.log(" paying back tokens " + addressTokens.map(token => " " + token.amount + "@" + token.symbol))
-        if (paybackTokens.length > 0) {
+    private async paybackTokenBalances(loanTokens: AddressToken[], collateralTokens: AddressToken[], telegram: Telegram, prevout: Prevout | undefined = undefined): Promise<boolean> {
+        
+        if(loanTokens.length == 0 && collateralTokens.length == 0) {
+            await telegram.send("ERROR: want to pay back, but nothing to do. please check logs")
+            console.error("no tokens to pay back or deposit")
+            return false
+        }
+        let waitingTx = undefined
+        if (loanTokens.length > 0) {
+            console.log(" paying back tokens " + loanTokens.map(token => " " + token.amount + "@" + token.symbol))
+            let paybackTokens: TokenBalance[] = []
+            loanTokens.forEach(addressToken => {
+                paybackTokens.push({ token: +addressToken.id, amount: new BigNumber(addressToken.amount) })
+            })
             const paybackTx = await this.paybackLoans(paybackTokens, prevout)
-
+            waitingTx= paybackTx
+            prevout= this.prevOutFromTx(paybackTx)
             await this.updateToState(ProgramState.WaitingForTransaction, VaultMaxiProgramTransaction.PaybackLoan, paybackTx.txId)
-            const success = await this.waitForTx(paybackTx.txId)
+        }
+        if(collateralTokens.length > 0) {
+            console.log(" depositing tokens " + collateralTokens.map(token => " " + token.amount + "@" + token.symbol))
+            for(const collToken of collateralTokens) {
+                const depositTx= await this.depositToVault(+collToken.id,new BigNumber(collToken.amount),prevout)
+                waitingTx= depositTx
+                prevout= this.prevOutFromTx(depositTx)
```

Comment:
in this case prevout receives the value of depositTx, but isn't used anymore, because deposit is the last in chain, correct?

Reply:
no, this is done in a loop (1 collateral at a time) so it could be (in theory) multiple collaterals being deposited here.